### ISSUE TITLE
Deploy a kubernetes cluster on AWS.

### ISSUE THREAD

**Issue by opadron** [spack/spack#7421]
This issue will track the tasks needed to prepare the infrastructure we'll need to support automated testing and provisioning of source and binary packages.

- [x] Acquire AWS account credentials
- [x] Setup spending alerts
- [x] Review Kubernetes setup documentation at https://kubernetes.io/docs/setup
- [x] Develop an initial EC2 host allocation based on installation requirements.
- [x] Deploy a minimal EC2 host allocation for development and testing.
- [x] Develop provisioning scripts and tools to automate the cluster setup process.
- [x] Deploy production allocation using developed scripts.
- [ ] Develop a retention policy for source packages.
- [ ] Develop a retention policy for binary packages.
- [ ] ? @tgamblin @scheibelp (setup build bot?)

The following issues should be considered blocked on this work:
#7121
#7123
#7116
#7117

**Comment by jcftang** [comment]
If you're interested in setting up a k8s cluster I'd suggest taking a look at kops - https://github.com/kubernetes/kops and also consider in just using AWS's offering of their managed k8s clusters.

**Comment by opadron** [comment]
Thanks for the info, @jcftang! We're still working out how we want to approach this and related work.  I will definitely give kops and AWS's managed solutions a look.

**Comment by opadron** [comment]
Hi, folks.  I wanted to provide an update on this effort:

Thank you, @jcftang, for suggesting kops for setting up Kubernetes on AWS.  We've started using kops and have had good initial success in setting up and configuring a basic cluster.

From the guidance from @tgamblin, we've decided not to directly approach the tasks of developing on a small deployment and then later migrating to a larger deployment.  Instead, we are continuing the work on the smaller cluster and plan to revisit the matter of scaling once we are able to demonstrate the full end-to-end solution for PR and release testing.

For similar reasons, we are also punting on formalizing any retention policies for source and binary packages.  Instead, we'll probably have the first few versions of the testing workflow just dump tarballs to an s3 bucket, and investigate matters relating to retention, package signing, and the like once we are able to upload packages and download them using spack as the client in testing.

At the moment, we are working on deploying the first set of services on the cluster.  Instead of build-bot, we are trying out Gitlab and the Gitlab Runner service for running the build jobs.  After this, we will add a dedicated CDash instance that the build jobs will report to, allowing us to track the status of each package over time.

Considering our progress, I had updated the to-do list above.  I propose that we close this issue (since I originally billed it as "Deploy Kubernetes" and we have indeed accomplished that much), and add new issues tracking each of the newly identified tasks that don't already have their own issue.  Then, we can further elaborate on the details of each task and go from there.

If this makes sense to others, I'll go ahead and make the changes.

cc @tgamblin
